File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
            stack = []
            hashmap = { ')':'(',
                       ']':'[', 
                      '}': '{' 
                      }
            for char in s:
                if char == '(' or char == '[' or char == '{':
                    stack.append(char)
                elif len(stack) != 0 and hashmap[char] == stack[-1]:
                    stack.pop()
                else:
                    return False
            return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        std::unordered_map<char, char>hashmap = {
            {']','['}, 
            {'}','{'}, 
            {')','('} 
        };
        std::stack<char> stack;
        for(char c : s){
            if (c == '[' || c == '{' || c == '(' ){
            stack.push(c);
        }
            // is the stack empty
            // what is the top
            // does my character correspond TO the top
            // last two can be helped with a hashmap because ican check if the key (right closing) is equal to top?
            else if ( !stack.empty() && (hashmap[c] == stack.top() )  ){
                stack.pop();
            }
            else{
                return false;
            }
        
   }
        
return stack.empty();
    }
};